pkg/utils/util.go

```diff
@@ -37,8 +39,17 @@ func GetToolchainClusterName(clusterType, serverAPIEndpoint, suffix string) (str
 	// in the original script.
 	fixedLength := len(clusterType) + len(suffix) + 1
 
-	maxAllowedClusterHostNameLen := 62 - fixedLength // I think 62 is here, because we might default the suffix to "1" later on
+	maxAllowedClusterHostNameLen := K8sLabelWithoutSuffixMaxLength - fixedLength // I think 62 is here, because we might default the suffix to "1" later on
+
+	clusterHostName, err := GetClusterHostName(serverAPIEndpoint, maxAllowedClusterHostNameLen, suffix)
+	if err != nil {
+		return "", err
+	}
 
+	return fmt.Sprintf("%s-%s", clusterType, clusterHostName), nil
+}
+
+func GetClusterHostName(serverAPIEndpoint string, maxAllowedClusterHostNameLen int, suffix string) (string, error) {
```

Comment:
This functionality was extracted to be reused somewhere else but in the end I switched on using the values from the validatedData so it's not actually required anymore, but still a small refactoring. I can change the function name so that it's private and not public or revert the change. It only makes a bit more clear on what that code does , at least for me.

Reply:
made private in https://github.com/kubesaw/ksctl/pull/53/commits/9742cf3f4f962b2c351d9b6095fe802b8bb1f515